[inductor] Fallback on complex64 kernels

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98158
* #97997
* #98120
* #98154
* #98150
* __->__ #98155
* #98121


Later PRs in this stack fixe graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

jansel committed:
[inductor] Fallback on complex64 kernels

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



The next PR in the stack fixes graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



The next PR in the stack fixes graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



The next PR in the stack fixes graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



The next PR in the stack fixes graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

ngimel commented on torch/_inductor/lowering.py:
```diff
@@ -1041,6 +1041,8 @@ def handler(*args, **kwargs):
 
 
 def unsupported_output_tensor(t: torch._subclasses.FakeTensor):
+    if t.dtype in (torch.complex32, torch.complex64):
```
```suggestion
    if t.dtype in (torch.complex32, torch.complex64, torch.complex128):
```

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



Later PRs in this stack fixe graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "[inductor] Fallback on complex64 kernels"



Later PRs in this stack fixe graph breaks in GoogleFnet which triggers errors from inductor trying to compile torch.complex64, this fixes that.


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel commented:
Note this PR seems to uncover another bug: https://github.com/pytorch/pytorch/issues/98175

jansel commented:
@pytorchbot merge
